### PR TITLE
Tcp connection

### LINKED ISSUE #2: Start SMTP server
Acceptance Criteria:
- Listens for traffic on certain port (1337)

### PR DESCRIPTION
**Pull request by charmalt**
closes #2

### CODE REVIEW

File: lib/tcpServer.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    this.port = port
+    this.address = address
+    this.clients = []
+  }
+
+  start () {
+    this.connection = net.createServer((socket) => {
+      let client = this.createClient(socket)
+      socket.on('data', (data) => {
+        // TBD integration with parser
```
Comment (JohnNewman1) [inline lib/tcpServer.js]: Is this where the handshake will go?

File: lib/tcpConnectedClient.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+class TCPConnectedClient {
+  constructor (socket) {
+    this.address = socket.remoteAddress
+    this.port = socket.remotePort
+    this.name = `${this.address}:${this.port}`
+    this.socket = socket
+  }
+
+  receiveMessage (message) {
```
Comment (nixlim) [inline lib/tcpConnectedClient.js]: Little bit confusing naming

File: package.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "private": true,
   "scripts": {
-    "test": "jest",
+    "test": "jest --forceExit",
```
Comment (nixlim) [inline package.json]: I would suggest that this is changed to:
jest --forceExit --coverage

File: lib/tcpConnectedClient.js (hunk 2 above)
Comment (georgesykes86) [reply lib/tcpConnectedClient.js]: We have discussed and agreed it is ok

File: lib/tcpServer.js (hunk 1 above)
Comment (charmalt) [reply lib/tcpServer.js]: yes. well done, John

**Comment by benfurber** [comment]
🚢 ----|>

File: package.json (hunk 3 above)
Comment (georgesykes86) [reply package.json]: Agreed and implemented

**Comment by georgesykes86** [comment]
:shipit:

**Pull request merged**